components/net/image_cache.rs

```diff
@@ -916,13 +1065,20 @@ impl ImageCache for ImageCacheImpl {
                 completed_loads: HashMap::new(),
                 placeholder_image,
                 placeholder_url,
-                compositor_api,
+                compositor_api: compositor_api.clone(),
                 vector_images: HashMap::new(),
                 rasterized_vector_images: HashMap::new(),
+                key_cache: KeyCache::new(),
+                pipeline_id,
             })),
             thread_pool: self.thread_pool.clone(),
         })
     }
+
+    fn fill_key_cache_with_batch_of_keys(&self, image_keys: Vec<WebRenderImageKey>) {
+        let mut store = self.store.lock().unwrap();
+        store.insert_keys_and_load_images(image_keys);
```

Comment:
As a note for a follow-up, ideally we should only insert keys while in the critical section, and finish the load outside of it(if possible).